aiogram/types/input_file.py

```diff
@@ -38,7 +38,12 @@ async def read(self, bot: "Bot") -> AsyncGenerator[bytes, None]:  # pragma: no c
 
 
 class BufferedInputFile(InputFile):
-    def __init__(self, file: bytes, filename: str, chunk_size: int = DEFAULT_CHUNK_SIZE):
+    def __init__(
+        self,
+        buffer_or_bytes: bytes | io.BufferedIOBase,
```

Comment:
The "file" argument should not be renamed because someone could use named arguments and this change breaks compatibility.

---

The `A | B` type hint is not compatible with Python 3.8.

---

`SrtingIO` is also a subclass of `BufferedIOBase`, but it is incompatible with the `InputFile.read` method, which should only generate chunks of bytes.